### PR TITLE
Added `gamma` to `LBFGSState`

### PR DESCRIPTION
**Pull request by zaccharieramzi**
This PR adds a new attribute to the `LBFGState` named tuple, `gamma`.
This attribute is needed when one wants to use the approximation of the Hessian inverse after the algorithm has run using `inv_hessian_product`.

Indeed I realized that in this implementation, the initial value of the approximation of the Hessian is recomputed at each iteration, as suggested by Nocedal (something that is not present in scipy to the best of my knowledge and understanding of Fortran).

The application I have in mind for the approximation of the Hessian inverse is https://arxiv.org/abs/2106.00553.

(I didn't write an issue first because it's such a small PR, and I was going to write this part anyway for my own xps)

### CODE REVIEW

**Comment by zaccharieramzi** [comment]
One aspect I am not too sure about though in the arguments of `inv_hessian_product` is `start`.
It indeed seems like it's used to perform the matrix multiplication in a certain order, but I don't understand why that's necessary.
Is it because with numerical errors the additions become non-commutative?

**Comment by mblondel** [comment]
> One aspect I am not too sure about though in the arguments of inv_hessian_product is start

We use a circular buffer to maintain the history without lists.

**Review by mblondel — APPROVED** [review]
LGTM, thanks a lot for the contribution

**Comment by zaccharieramzi** [comment]
I think I don't have access to the logs of the copybara failure, or at least I am not sure how to access them in order to correct the failure.

File: jaxopt/_src/lbfgs.py [hunk 1]
```diff
@@ -142,6 +142,7 @@ class LbfgsState(NamedTuple):
   s_history: Any
   y_history: Any
   rho_history: jnp.ndarray
+  gamma: Any = jnp.array(1.0)
```
[the remark this answers is not in the document]
Comment (mblondel) [reply jaxopt/_src/lbfgs.py:145]: Sorry I meant
```python
  gamma: jnp.ndarray
```

[the remark this answers is not in the document]
Comment (zaccharieramzi) [reply jaxopt/_src/lbfgs.py]: done

**Pull request merged**